db/migrations/202407012100_transactions.go

```diff
@@ -0,0 +1,59 @@
+package migrations
+
+import (
+	_ "embed"
+
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+// This migration
+// - Replaces the old payments table with a new transactions table
+// - Adds new properties to apps
+//   - isolated boolean
+//
+// - Renames max amount on app permissions to be clear its in sats
+var _202407012100_transactions = &gormigrate.Migration{
+	ID: "202407012100_transactions",
+	Migrate: func(tx *gorm.DB) error {
+
+		if err := tx.Exec(`
+CREATE TABLE transactions(
+	id integer PRIMARY KEY AUTOINCREMENT,
+	app_id integer,
+	request_event_id integer,
+	type text,
+	state text,
+	payment_request text,
+	preimage text,
+	payment_hash text,
+	description text,
+	description_hash text,
+	amount_msat integer,
+	fee_msat integer,
+	fee_reserve_msat integer,
+	created_at datetime,
+	updated_at datetime,
+	expires_at datetime,
+	settled_at datetime,
+	metadata text,
+	self_payment boolean
```

Comment:
we probably should have an index on 
+ app_id
+ state
+ type
+ payment_hash
+ created_at / settled_at? 

need to check what queries we make, but I think we need something here. 
for example for this a combined might be helpful

`"app_id = ? AND type = ? AND (state = ? OR state = ?) AND created_at > ?"`
`app_id = ? AND type = ? AND state = ?`


make sure `type` is no reserved word.

Reply:
why would you want to wait? what is an argument of not adding an index?
you know which queries you do and I am not sure if we can identify slow queries.